Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
-**Content**
+Getting Started
+===============
 
-1. [Simple example]
+This project tries to be consistent with [ReactiveX.io](http://reactivex.io/). The general cross platform documentation and tutorials should also be valid in case of `RxSwift`.
 
-## Simple example
+1. [Observables aka Sequences](#observables-aka-sequences)
+1. [Creating your own `Observable` (aka sequence producers)](#creating-your-own-observable-aka-sequence-producers)
+1. [Operators](#operators)
+1. [Error handling](#error-handling)
+1. [Debugging](#debugging)
+1. [UI layer tips](#ui-layer-tips)
+1. [Examples](Examples.md)
 
-To understand these examples, you will need to understand to `>-` operator.
+# Observables aka Sequences
 
-This is the definition of `>-` operator
+## Basics
+This equivalence is the most important thing to understand.
+
+Sequences are a simple concept.
+
+Everybody is familiar with sequences. Lists/sequences are probably one of the first concepts mathematicians/programmers learn.
+
+Here is a sequence of numbers
+
+
+```
+--1--2--3--4--5--6--| // it terminates normally
+```
+
+Here is another one with characters
+
+```
+--a--b--a--a--a---d---X // it terminates with error
+```
+
+Some sequences are finite, and some are infinite, like sequence of button taps
+
+```
+---tap-tap-------tap--->
+```
+
+These diagrams are called marble diagrams.
+
+[http://rxmarbles.com/](http://rxmarbles.com/)
+
+If everything is a sequence and every operation is just a transformation of input sequence into output sequence then it's pretty straightforward to compose operations.
+
+The concept of sequences lifts a lot of cognitive overload. You stop worrying about asynchronous events and how to compose them and think in terms of synchronous operations over sequences.
+
+People are creatures with huge visual cortexes. When you can visualize something easily, it's a lot easier to reason about it.
+
+If we were to specify sequence grammar as regular expression it would look something like this
+
+**Next* (Error | Completed)**
+
+This describes the following:
+
+* sequences can have 0 or more elements
+* once `Error` or `Completed` event is received sequence can't produce any other element
```

Comment:
> once `Error` or `Completed` event is received sequence can't produce any other element

once an `Error` or `Completed` event is received, the sequence can't produce any other element